addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -364,3 +365,26 @@ def _get_default_ab_testing_campaign_values(self, values=None):
                 campaign_values['name'] = _("A/B Test: %s", sms_subject)
             campaign_values['ab_testing_sms_winner_selection'] = self.ab_testing_sms_winner_selection
         return campaign_values
+
+    # --------------------------------------------------
+    # MAIL OVERRIDE
+    # --------------------------------------------------
+
+    @api.model
+    def _sort_activities_by_record_by_module(self, activities):
+        """ Override to separate mailing.mailing related activities between mass_mailing
+            and mass_mailing_sms modules depending on their mailing_type field value.
+        """
+        activities_by_rec_id_by_module = defaultdict(lambda: defaultdict(lambda: self.env["mail.activity"]))
+        group_custom_info = defaultdict(lambda: defaultdict(lambda: False))
+        module_sms = 'mass_mailing_sms'
+        group_custom_info[module_sms] = {'name': _('SMS Marketing'), 'domain': [['mailing_type', '=', 'sms']]}
+        module_email = 'mass_mailing'
+        group_custom_info[module_email] = {'name': _('Email Marketing'), 'domain': [['mailing_type', '!=', 'sms']]}
+        for activity in activities:
+            record = self.env[activity.res_model].browse(activity.res_id)
+            if record.mailing_type == 'sms':
+                activities_by_rec_id_by_module[module_sms][activity.res_id] += activity
+            else:
+                activities_by_rec_id_by_module[module_email][activity.res_id] += activity
+        return activities_by_rec_id_by_module, group_custom_info
```

Comment:
I haven't looked at the code in detail, but it's not great to not see any super call when parent implementation is not trivial and this one neither is, without any `super()` call. There's likely some code duplication and/or hidden coupling of both implementation.

It would be much better to see super calls and/or override only a very specific function, so the override is minimal and easy to understand

Reply:
Indeed it is a bit weird ... could probably test the email / sms marketing split in test_mass_mailing module, as both are in dependencies.

Reply:
Following on this, it looks like `group_custom_info` and `activities_by_rec_id_by_module` are quite independent and could be two different functions.

For example the override in mailing does not touch `activities_by_rec_id_by_module`, while the override in partner leaves `group_custom_info`.

`group_custom_info` looks like it returns static values for each models and does not depend on the param (activities).

Reply:
Thank you for your comments and suggestions, I've rewritten that part to take them into account. So know, the method `systray_get_activities` of res.users will use:
- `_sort_activities_by_record` of BaseModel to sort the activities with a given res_model
- `_get_systray_group_custom_infos` of res.users to get customized name/domain for activity groups
It also does less request than my previous implementation. Let me know if you have other comment about that.